src/config/HYPREConfig.cmake.in

```diff
@@ -17,29 +29,37 @@ set(HYPRE_ENABLE_HYPRE_BLAS @HYPRE_ENABLE_HYPRE_BLAS@)
 set(HYPRE_ENABLE_HYPRE_LAPACK @HYPRE_ENABLE_HYPRE_LAPACK@)
 set(HYPRE_ENABLE_PERSISTENT_COMM @HYPRE_ENABLE_PERSISTENT_COMM@)
 set(HYPRE_ENABLE_FEI @HYPRE_ENABLE_FEI@)
-set(HYPRE_WITH_MPI @HYPRE_WITH_MPI@)
-set(HYPRE_WITH_OPENMP @HYPRE_WITH_OPENMP@)
-set(HYPRE_WITH_HOPSCOTCH @HYPRE_WITH_HOPSCOTCH@)
-set(HYPRE_USING_DSUPERLU @HYPRE_USING_DSUPERLU@)
-set(HYPRE_USING_MAGMA @HYPRE_USING_MAGMA@)
-set(HYPRE_WITH_CALIPER @HYPRE_WITH_CALIPER@)
-set(HYPRE_PRINT_ERRORS @HYPRE_PRINT_ERRORS@)
-set(HYPRE_TIMING @HYPRE_TIMING@)
-set(HYPRE_BUILD_EXAMPLES @HYPRE_BUILD_EXAMPLES@)
-set(HYPRE_BUILD_TESTS @HYPRE_BUILD_TESTS@)
-set(HYPRE_USING_HOST_MEMORY @HYPRE_USING_HOST_MEMORY@)
-set(HYPRE_WITH_CUDA @HYPRE_WITH_CUDA@)
-set(HYPRE_WITH_SYCL @HYPRE_WITH_SYCL@)
+set(HYPRE_ENABLE_MPI @HYPRE_ENABLE_MPI@)
```

Comment:
Why should MPI use "enable"?  It's an external (optional) package.

Reply:
Thanks, Rob! Now I'm actually in doubt about this. Should we also use "with" for the OpenMP, CUDA, HIP and SYCL options since they involve external (optional) packages? i.e., `libomp`, `libcuda`, `libamdip64`, `libsycl`...

Reply:
I was wondering the same thing, though we used enable on the autotools side.  I never liked the fact that there are two words to choose from for options.  It doesn't seem to provide any real benefit.  I wonder if people even pay attention to this much anymore.  Since cmake doesn't promote these rules (correct me if I am wrong on that), we can do whatever we like here.  Also, since our autotools build system is more widely used, we have an opportunity to define the cmake side of things without affecting too many users.

Reply:
Hi Rob, I cannot find an official recommendation about this in cmake's documentation and I agree we could just use one of them, i.e., "with" or "enable". If you are fine with it, I can replace all remaining "with" occurrences with "enable". Looking at what other LLNL packages do, it seems "enable" is more widely used

Reply:
Okay, then let's stick with the xsdk recommendations and try to sort out when to use "with" vs "enable".  @oseikuffuor1 , please do keep chiming in.  Thanks!

Reply:
Thanks @oseikuffuor1 , do you recall the policy number for this recommendation?

Reply:
@balay could you chime in this conversation? We are deciding whether to use "HYPRE_WITH", "HYPRE_ENABLE" or both for external package dependencies in our cmake build system

Reply:
@victorapm it was one of the initial documents written on the build system policies. I am not sure where that is hosted now, but I'll see if I can dig up a local copy. Hopefully Satish or @ulrikeyang can provide some input/ guidance here. Otherwise I'm fine with also keeping consistency between Autotools and cmake.

Reply:
Thanks @oseikuffuor1! I'm curious to know if this ended up being included in the most recent set of policies. I searched for it but couldn't find it https://github.com/xsdk-project/xsdk-community-policies

Reply:
I poked around on the xsdk site as well, and I can't find a recommendation for cmake naming in any of the releases.  In version 0.5 and before, either an autotools or cmake style build system interface was required, but that was removed in favor of simply requiring spack.  Maybe that's what you are remembering @oseikuffuor1 ?

Assuming xsdk does not require a specific convention here, I'm still undecided about whether we should continue to mimic the autotools naming or just throw it out in favor of always using "enable" for simplicity (cmake only).

Reply:
Thanks, Rob! I was initially thinking to mimic autotools, but after thinking a bit more about this, I can't see a strong benefit of having both forms. It's also not trivial to decide which one to use in certain situations. So, I support using just "enable"

Reply:
Looks like xsdk policy  was changed to be more spack centric at:

https://github.com/xsdk-project/xsdk-community-policies/commit/663feada8bc5b6d0a80ecf7f6a94897134c7c3ff

wrt petsc [we don't have cmake interface, and likely don't follow autoconf convention] we use `--with-` for petsc features as well as external pkgs.
```
  --with-zlib=<bool>
  --with-zlib-dir=<dir>
  --with-zlib-pkg-config=<dir>
  --with-zlib-include=<dirs>
  --with-zlib-lib=<libraries: e.g. [/Users/..../libzlib.a,...]>
```
```
  --with-precision=<__fp16,single,double,__float128>
```

wrt MPI/CUDA/SYCL etc - there are 2 components here - one is compilers - the other is library. So we have - for ex. with CUDA
```
  --with-cudac=<prog> - i.e. nvcc
```
and
```
  --with-cuda=<bool>
  --with-cuda-dir=<dir>
  --with-cuda-include=<dirs>
  --with-cuda-lib=<libraries: e.g. [/Users/..../libcuda.a,...]>
  --with-cuda-arch
```
etc..

cc: @barrysmith

Reply:
Adding to this, it looks like SUNDIALS uses "enable" pretty exclusively in their cmake build system.  I think I am leaning towards going with the "enable" option, but I'd like to hear other opinions if people are around and available to comment.  @oseikuffuor1 @ulrikeyang @waynemitchell @liruipeng

Reply:
Other LLNL packages also using "enable": umpire, chai, camp, raja, caliper, tribol, axom, kripke, silo, samrai, adiak

Reply:
It does look like the initial policies changed in favor of supporting spack. And it looks like PETSC used "with" for their options with their native build system. I'm fine with using "enable" like the rest of the libraries here (as @rfalgout and @victorapm noted).